FiveCalls/FiveCalls/ScheduleReminders.swift

```diff
@@ -0,0 +1,285 @@
+                        existingSelectedTime = triggerDate
+                        selectedTime = triggerDate
+                    }
+                }
+                .onChange(of: remindersEnabled) { newValue in
+                    if newValue {
+                        requestNotificationAccess()
+//                    } else {
+//                        clearNotifications()
+                    }
+                }
+//                .onChange(of: selectedTime) { newValue in
+//                    if newValue != existingSelectedTime {
+//                        clearNotifications()
+//                    }
+//                }
+//                .onChange(of: selectedDayIndices) { newValue in
+//                    if newValue != existingSelectedDayIndices {
+//                        clearNotifications()
+//                    }
+//                }
+            }
+        }
+    }
+    
+    private func requestNotificationAccess() {
+        let options: UNAuthorizationOptions = [.alert, .sound, .badge];
+        UNUserNotificationCenter.current().requestAuthorization(options: options) { (success, error) in
+            AnalyticsManager.shared.trackEventOld(withName: "Notification Access", andProperties: ["success": "\(success)"])
+        }
+    }
+    
+    private func clearNotifications() {
+        UNUserNotificationCenter.current().removeAllPendingNotificationRequests()
+    }
+    
+    private func onDismiss() {
+        let cannotDismiss = selectedDayIndices.isEmpty && remindersEnabled
+        if cannotDismiss {
+            shouldShake = true
+        } else if existingSelectedTime == selectedTime &&
+                    existingSelectedDayIndices == selectedDayIndices && remindersEnabled {
+            dismiss()
+        } else {
+            clearNotifications()
+            if remindersEnabled {
+                for index in selectedDayIndices {
+                    let notificationContent = notificationContent()
+                    let notificationTrigger = notificationTrigger(date: selectedTime, dayIndex: index)
+                    let request = UNNotificationRequest(identifier: "5calls-reminder-\(index)", content: notificationContent, trigger: notificationTrigger)
+                    UNUserNotificationCenter.current().add(request)
+                }
+            }
+
+            dismiss()
+        }
+    }
+    
+    private func indices(from notifications: [UNNotificationRequest]) -> [Int] {
+        let calendar = Calendar(identifier: .gregorian)
+        return notifications.compactMap({ notification in
+            if let calendarTrigger = notification.trigger as? UNCalendarNotificationTrigger {
+                return calendar.component(.weekday, from: (calendarTrigger.nextTriggerDate()!))
+            }
+            
+            return nil
+        })
+    }
+    
+    private func notificationContent() -> UNMutableNotificationContent {
```

Comment:
TODO: split out some of these duplicate functions into helpers used by non-SwiftUI version.